electrum/lnsweep.py

```diff
@@ -461,19 +567,97 @@ def create_htlctx_that_spends_from_our_ctx(
+        wallet = chan.lnworker.wallet
+        coins = wallet.get_spendable_coins(None)
+
+        def fee_estimator(size):
+            if htlc_direction == SENT:
+                # we deal with an offered HTLC and therefore with a timeout transaction
+                # in this case it is not time critical for us to sweep unless we
+                # become a forwarding node
+                fee_per_kb = wallet.config.eta_target_to_fee(HTLC_TRANSACTION_SWEEP_TARGET)
+            else:
+                # in the case of a received HTLC, if we have the hash preimage,
+                # we should sweep before the timelock expires
+                expiry_height = htlc.cltv_expiry
+                current_height = wallet.network.blockchain().height()
+                deadline_blocks = expiry_height - current_height
+                # target block inclusion with a safety buffer
+                target = int(deadline_blocks / HTLC_TRANSACTION_DEADLINE_FRACTION)
+                fee_per_kb = wallet.config.eta_target_to_fee(target)
+            if not fee_per_kb:  # testnet and other cases
+                fee_per_kb = wallet.config.fee_per_kb()
+            fee = wallet.config.estimate_fee_for_feerate(fee_per_kb=fee_per_kb, size=size)
+            # we only sweep if it is makes sense economically
+            if fee > htlc.amount_msat // 1000:
+                raise UneconomicFee
+            return fee
+
+        coin_chooser = coinchooser.get_coin_chooser(wallet.config)
+        change_address = wallet.get_single_change_address_for_new_transaction()
+        funded_htlc_tx = coin_chooser.make_tx(
+            coins=coins,
+            inputs=maybe_zero_fee_htlc_tx.inputs(),
+            outputs=maybe_zero_fee_htlc_tx.outputs(),
+            change_addrs=[change_address],
+            fee_estimator_vb=fee_estimator,
+            dust_threshold=wallet.dust_threshold())
+
+        # place htlc input/output at corresponding indices (due to sighash single)
+        htlc_outpoint = TxOutpoint(txid=bfh(ctx.txid()), out_idx=ctx_output_idx)
+        htlc_input_idx = funded_htlc_tx.get_input_idx_that_spent_prevout(htlc_outpoint)
+
+        htlc_out_address = maybe_zero_fee_htlc_tx.outputs()[0].address
+        htlc_output_idx = funded_htlc_tx.get_output_idxs_from_address(htlc_out_address).pop()
+        inputs = funded_htlc_tx.inputs()
+        outputs = funded_htlc_tx.outputs()
+        if htlc_input_idx != 0:
+            htlc_txin = inputs.pop(htlc_input_idx)
+            inputs.insert(0, htlc_txin)
+        if htlc_output_idx != 0:
+            htlc_txout = outputs.pop(htlc_output_idx)
+            outputs.insert(0, htlc_txout)
+        final_htlc_tx = PartialTransaction.from_io(
+            inputs,
+            outputs,
+            locktime=maybe_zero_fee_htlc_tx.locktime,
+            version=maybe_zero_fee_htlc_tx.version,
+            BIP69_sort=False
+        )
+
+        # sign the fee input(s): TODO: how to deal with password-protected wallets?
```

Comment:
I see this is not using `wallet.wallet_password` yet

Reply:
Somehow got lost during rebasing :+1: